hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/v2/components/tables/datanodesTable.tsx

```diff
@@ -0,0 +1,315 @@
+              replicationFactor={pipeline.replicationFactor}
+              replicationType={pipeline.replicationType}
+              leaderNode={pipeline.leaderNode}
+              isLeader={pipeline.leaderNode === record.hostname} />
+            {pipeline.pipelineID}
+          </div >
+        ))
+      }
+
+      return (
+        <Popover
+          content={
+            renderPipelineIds(pipelines)
+          }
+          title="Related Pipelines"
+          placement="bottomLeft"
+          trigger="hover">
+          <strong>{pipelines.length}</strong> pipelines
+        </Popover>
+      );
+    }
+  },
+  {
+    title: () => (
+      <span style={headerIconStyles} >
+        Leader Count
+        <Tooltip
+          title='The number of Ratis Pipelines in which the given datanode is elected as a leader.' >
+          <InfoCircleOutlined style={{ paddingLeft: '4px' }} />
+        </Tooltip>
+      </span>
+    ),
+    dataIndex: 'leaderCount',
+    key: 'leaderCount',
+    sorter: (a: Datanode, b: Datanode) => a.leaderCount - b.leaderCount
+  },
+  {
+    title: 'Containers',
+    dataIndex: 'containers',
+    key: 'containers',
+    sorter: (a: Datanode, b: Datanode) => a.containers - b.containers
+  },
+  {
+    title: () => (
+      <span style={headerIconStyles}>
+        Open Container
+        <Tooltip title='The number of open containers per pipeline.'>
+          <InfoCircleOutlined style={{ paddingLeft: '4px' }} />
+        </Tooltip>
+      </span>
+    ),
+    dataIndex: 'openContainers',
+    key: 'openContainers',
+    sorter: (a: Datanode, b: Datanode) => a.openContainers - b.openContainers
+  },
+  {
+    title: 'Version',
+    dataIndex: 'version',
+    key: 'version',
+    sorter: (a: Datanode, b: Datanode) => a.version.localeCompare(b.version),
+    defaultSortOrder: 'ascend' as const
+  },
+  {
+    title: 'Setup Time',
+    dataIndex: 'setupTime',
+    key: 'setupTime',
+    sorter: (a: Datanode, b: Datanode) => a.setupTime - b.setupTime,
+    render: (uptime: number) => {
+      return uptime > 0 ? moment(uptime).format('ll LTS') : 'NA';
+    }
+  },
+  {
+    title: 'Revision',
+    dataIndex: 'revision',
+    key: 'revision',
+    sorter: (a: Datanode, b: Datanode) => a.revision.localeCompare(b.revision),
+    defaultSortOrder: 'ascend' as const
+  },
+  {
+    title: 'Build Date',
+    dataIndex: 'buildDate',
+    key: 'buildDate',
+    sorter: (a: Datanode, b: Datanode) => a.buildDate.localeCompare(b.buildDate),
+    defaultSortOrder: 'ascend' as const
+  },
+  {
+    title: 'Network Location',
+    dataIndex: 'networkLocation',
+    key: 'networkLocation',
+    sorter: (a: Datanode, b: Datanode) => a.networkLocation.localeCompare(b.networkLocation),
+    defaultSortOrder: 'ascend' as const
+  }
+];
+
+const DatanodesTable: React.FC<DatanodeTableProps> = ({
+  data,
+  handleSelectionChange,
+  decommissionUuids,
+  selectedColumns,
+  loading = false,
+  selectedRows = [],
+  searchColumn = 'hostname',
```

Comment:
This issue is present in both the old and new UI: if a user accidentally adds a space at the end of their search input, the search doesn’t work. While it’s not a critical issue, we could remove any unnecessary spaces from the search input automatically. It’s a small improvement, but I noticed it when I was testing the search functionality.

This issue is present in both the old and new UI: if a user accidentally adds a space at the end of their search input, the search doesn’t work. While it’s not a critical issue, we could remove any unnecessary spaces from the search input automatically. It’s a small improvement, but I noticed it when I was testing the search functionality.

This issue is present in both the old and new UI: if a user accidentally adds a space at the end of their search input, the search doesn’t work. While it’s not a critical issue, we could remove any unnecessary spaces from the search input automatically. It’s a small improvement, but I noticed it when I was testing the search functionality.

<img width="1016" alt="image" src="https://github.com/user-attachments/assets/9566f1bc-f1d6-466f-8e55-258420a200e2">